core/trino-spi/src/main/java/io/trino/spi/type/ShortDecimalType.java

```diff
@@ -127,7 +127,7 @@ private static boolean equalOperator(long left, long right)
     @ScalarOperator(HASH_CODE)
     private static long hashCodeOperator(long value)
     {
-        return value;
+        return AbstractLongType.hash(value);
```

Comment:
LGTM

btw i don't know why for some types we have both HASH_CODE and XX_HASH_64. It looks like removing `hashCodeOperator` from here would also solve the problem